src/sync_manager.hpp

```diff
@@ -58,39 +66,32 @@ friend struct SyncSession;
     void set_log_level(util::Logger::Level) noexcept;
     void set_logger_factory(SyncLoggerFactory&) noexcept;
     void set_error_handler(std::function<sync::Client::ErrorHandler>);
-    void set_login_function(SyncLoginFunction);
+    void set_bind_session_function(SyncBindSessionFunction);
 
     /// Control whether the sync client attempts to reconnect immediately. Only set this to `true` for testing purposes.
     void set_client_should_reconnect_immediately(bool reconnect_immediately);
     /// Control whether the sync client validates SSL certificates. Should *always* be `true` in production use.
     void set_client_should_validate_ssl(bool validate_ssl);
 
-    std::shared_ptr<SyncSession> get_session(const std::string& path, const SyncConfig& config);
-    std::shared_ptr<SyncSession> get_existing_active_session(const std::string& path) const;
+    // Get a sync user for a given identity, or create one if none exists yet, and set its token.
+    // If a logged-out user exists, it will marked as logged back in.
+    shared_ptr<SyncUser> get_user(std::string identity, std::string user_token, bool is_admin=false);
```

Comment:
I think it's confusing to make references to a new system that has not yet been designed. The code should reflect the current design, unless we have a very clear idea of what we think the future will look like.

I was under the impression that with the current implementation it's not necessary to have any of the auth-related information in order to start a sync session, with it only being necessary at the point the log-in callback is called. This new design appears to require it up front. Am I missing something about how things currently work?